### PR TITLE
feat: better build system

### PR DESCRIPTION
**Pull request by theomonnom**
- windows arm64 is ignored
- musllinux is also ignored

download prebuilt ffi library from the Rust repo at build time
also build platform wheel to reduce binary size

### CODE REVIEW

**Review by davidzhao — APPROVED** [review]
lgtm!

how does it handle the upload to pypi part?

File: setup.py [hunk 1]
```diff
@@ -12,40 +12,85 @@
...
+        'bdist_wheel': bdist_wheel,
+        'build_py': BuildPyCommand,
+    },
     classifiers=[
-        "Development Status :: 3 - Alpha",
         "Intended Audience :: Developers",
         "License :: OSI Approved :: Apache Software License",
+        "Topic :: Multimedia :: Sound/Audio",
+        "Topic :: Multimedia :: Video",
+        "Topic :: Scientific/Engineering :: Artificial Intelligence",
```
Comment (davidzhao) [inline setup.py:77]: 🥇

File: setup.py [hunk 2]
```diff
@@ -12,40 +12,85 @@
...
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3 :: Only",
     ],
-
-    keywords="webrtc, livekit",
+    keywords=["webrtc", "realtime", "audio", "video", "livekit"],
     license="Apache-2.0",
     packages=["livekit"],
-    python_requires=">=3.7, <4",
+    python_requires=">=3.7.0",
```
Comment (davidzhao) [inline setup.py:88]: do we really need 3.7? or is 3.6 ok too?

**Review by davidzhao — APPROVED** [review]
🚀

File: setup.py (hunk 2 above)
Comment (theomonnom) [reply setup.py:88]: We use dataclass, I just saw this is from Python 3.7: https://peps.python.org/pep-0557/

**Comment by theomonnom** [comment]
The upload part will be inside another PR

**Pull request merged**